### ISSUE TITLE
Unable to run on up to date Arch/no speech

### ISSUE THREAD

**Issue by celtichawk** [odilia-app/odilia#45]
I am running the latest Arch Linux, with every single update. Running the AUR build gives me several errors

1. It does not move the config.toml into the .config/odilia directory
2. Moving that file into the correct place results in Odilia starting, with no speech and a flood of debug events handled withhout error events
3. Building from source and running the setup permissions script installs, and works fine however, Odilia has zero speech at all, as well as the aforementioned errors
I have run with Xorg and Sway, with zero luck running Odilia at all, it is acting as if speech-dispatcher is not picking up Odilia, despite it starting up fine, connecting with speech-dispatcher.

I figured maybe Orca was causing that so I pkilled Orca, with no luck at all. Odilia refuses to speak at all.

Things I've tried

Install from AUR, and remove,then remove the cache and start over
Manually create the config file in .config/Odilia
Build from source and run from both a hand made .desktop file, as well as a terminal with no luck

All I can do is ctrl+C Odilia from a terminal

**Comment by TTWNO** [comment]
The AUR package is not updated to reflect the new architecture of Odilia yet. That's on me since I manage the AUR package.

I'll create an issue for this ASAP and link it here.

**Comment by TTWNO** [comment]
https://github.com/odilia-app/odilia/issues/46

**Comment by celtichawk** [comment]
Ah, are there plans to get an updated AUR package, or plans for an updated install doc? I've manually installed it, somehow and am still getting zero speech at all, even after a fresh clone, running the required script from the readme, and doing ./Odilia, nothing at all

So what am I missing then? I've been able to build it from the repo. I've been able to get it to run, and connect to speech-ispatcher, with nothing past that

**Comment by TTWNO** [comment]
If it's connected to speech-dispatcher, then it should start speaking soon after opening an accessible application like Firefox. Can you send the debug log here? You may want to obscure some paths if they're unique to your system of course.

**Comment by celtichawk** [comment]
Where would I find the debug log at?

**Comment by DataTriny** [comment]
@celtichawk debug logs are printed on the standard output by default. So just copy/paste the output you get on your terminal while Odilia is running.

**Comment by celtichawk** [comment]
All I've been getting is debug event handled without error, just that one line over, and over, and over again without a second or third line or any indication of what the exact debug issue is

**Comment by albertotirla** [comment]
open another application, like go in your mate menu, gnome shell or whatever and pick an application you know it's accessible. If that still breaks, report it here. Also, maybe worth looking at the speech dispatcher logs for sections containing odilia?

**Comment by celtichawk** [comment]
The speech is there, however it is drowned out by the constant printing of errors to the console, the stream of debug event caught without error, so yes there is speech, but there is something going on with that. I had to run Odilia from a run box to avoid the terminal spam.

I'm not seeing anything too out of the ordinary in the speech-dispatcher logs however so I'm at a loss as to why it's acting up, but there is speech, as long as in my case I'm not running it in a console

**Comment by TTWNO** [comment]
Ok, yes you're right. We need to get the control character to stop speech ASAP. The reason I haven't noticed this bug is because my terminal emulator (`foot`), is not accessible, so no information is sent to Odilia when it starts.

**Comment by albertotirla** [comment]
OK, that's easily fixable, but it depends on how we want to handle logging. We could either pipe it to a file in a system location best suited for this thing, rely on the fact that odilia will be run in the context of systemd anyway and pipe to stdout/stderr knowing that systemd will transfer that to the journal, or try to do both and possibly fail. But yes, the previous prototype had this issue as well, however due to how the code was structured then, we couldn't slot in this change so readily...well, now we can.

**Comment by TTWNO** [comment]
The latesr version of Odilia should have this issue solved. Please let me kmow.

**Comment by TTWNO** [comment]
@celtichawk , can you confirm that this issue is fixed?

The latest commit has a script: `./scripts/odilia` which you can use to start Odilia. It should launch without any debug info, unless there is some sort of fatal error and it exits.

**Comment by celtichawk** [comment]
Yes, it's fixed now, sorry or taking so long to reply, can go ahead and close this now
